utility/utility.go

```diff
@@ -15,8 +15,20 @@ func getXcodeVersionFromXcodebuildOutput(outStr string) (models.XcodebuildVersio
 		return models.XcodebuildVersionModel{}, fmt.Errorf("failed to parse xcodebuild version output (%s)", outStr)
 	}
 
-	xcodebuildVersion := split[0]
-	buildVersion := split[1]
+	firstLineIndex := -1
+	for i, line := range split {
+		if strings.HasPrefix(line, "Xcode ") {
+			firstLineIndex = i
+			break
+		}
+	}
+
+	if firstLineIndex < 0 {
+		return models.XcodebuildVersionModel{}, fmt.Errorf("couldn't find Xcode version in output: %s", outStr)
+	}
```

Comment:
The context is not immediately visible when I am reading these lines. Looking at the test definitely help explaining why this is needed. What if we move this logic into a separate method with a more explanatory name like
```
func filterXcodeWarningsFromTheOutputAsAWorkaround(input []string) ([]string, error) {
    ...
}
```
and then at the call site we would only have 
```
split, err := filterXcodeWarningsFromTheOutputAsAWorkaround(split)
if err != nil {
    return models.XcodebuildVersionModel{}, err
}
```
This way the original method maybe remains more readable.

Reply:
Thanks, I also like this style better. Please take a look now.

Reply:
Looks great.